### PR TITLE
Fix arrange mailbox in alphabet

### PR DESCRIPTION
**Pull request by hoangdat**
Screen Shot 2021-08-31 at 18 04 04

@dab246 please review this PR.

### CODE REVIEW

**Comment by chibenwa** [comment]
Are mailbox of a sub-level sorted?

File: lib/features/mailbox/domain/extensions/mailbox_name_extension.dart [hunk 1]
```diff
@@ -1,5 +1,10 @@
 import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
 
 extension MailboxNameExtension on MailboxName {
-  int compareToSort(MailboxName mailboxName) => name.compareTo(mailboxName.name);
+  int compareAlphabetically(MailboxName? other) {
+    if (other == null) {
+      return 1;
```
Comment (dab246) [inline lib/features/mailbox/domain/extensions/mailbox_name_extension.dart:6]: why return `1` not `0`

File: lib/features/mailbox/presentation/mailbox_controller.dart [hunk 2]
```diff
@@ -87,7 +88,11 @@ class MailboxController extends BaseController {
   void onError(error) {}
 
   void _buildTree(List<PresentationMailbox> folderMailboxList) async {
-    folderMailboxTree.value = await _treeBuilder.generateMailboxTree(folderMailboxList);
+    folderMailboxTree.value = await _treeBuilder.generateMailboxTree(folderMailboxList)
+        .then((tree) {
+          tree.root.childrenItems?.sort((node, other) => node.item.name?.compareAlphabetically(other.item.name) ?? -1);
```
Comment (dab246) [inline lib/features/mailbox/presentation/mailbox_controller.dart:93]: why  default `-1`

Comment (hoangdat) [reply lib/features/mailbox/presentation/mailbox_controller.dart:93]: if this node is `null`, it should be below `other`

File: lib/features/mailbox/domain/extensions/mailbox_name_extension.dart (hunk 1 above)
Comment (hoangdat) [reply lib/features/mailbox/domain/extensions/mailbox_name_extension.dart:6]: if `other` is null, it should be less than `node`

**Comment by dab246** [comment]
Child node in sub-tree has not been sorted yet

Simulator Screen Shot - iPhone 12 Pro Max - 2021-09-01 at 09 22 29

**Comment by chibenwa** [comment]
> Child node in sub-tree has not been sorted yet

I think we need to, no?

**Comment by hoangdat** [comment]
> > Child node in sub-tree has not been sorted yet
> 
> I think we need to, no?

yes, I am fixing up.

**Comment by hoangdat** [comment]
image

**Review by dab246 — APPROVED** [review]
Can be merged.

**Pull request merged**